### PR TITLE
Polymer 3 2018 12 31 132251

### PR DESCRIPTION
**Pull request by mdulat**

### CODE REVIEW

File: d2l-datetime-picker.js [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+				flex: 1 0 6rem;
+				max-width: 7rem;
+				padding-right: 0.75rem;
+			}
+
+			.d2l-date-picker-container {
+				flex: 1 1 auto;
+			}
+
+			:host-context([dir="rtl"]) .d2l-time-picker-container {
```
Comment (dbatiste) [inline d2l-datetime-picker.js]: Looks like this needs to be changed to `:host(:dir(rtl))` for Polymer 3.  I think this particular selector may actually show up in LTR on Edge.

File: d2l-datetime-picker.js [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+			@media (max-width: 615px), (max-device-width: 960px) {
+				label {
+					font-size: 0.6rem;
+					padding-right: var(--d2l-datetime-picker-label-padding);
+					text-align: right;
+					@apply --d2l-datetime-picker-label-mobile;
+				}
+
+				:host-context([dir="rtl"]) label {
```
Comment (dbatiste) [inline d2l-datetime-picker.js]: This one will need fixing too I think.

**Comment by mdulat** [comment]
> ;

moment is an es6 module, but the problem is that moment-timezone is not. Importing them this way gives a console error `cannot read property 'moment' of undefined`, referring to https://github.com/moment/moment-timezone/blob/develop/builds/moment-timezone-with-data.js#L16

**Comment by awikkerink** [comment]
`import 'moment/min/moment.min.js';` doesn't add moment to the window object?

**Comment by mdgbayly** [comment]
The sense I got from Margaret on Friday was that some of the amd/commonjs module plumbing was breaking things when trying to use these libs as es6 imports but with the intention of setting up browser globals. But if it is just that window.moment is not defined when trying to import moment-timezone...

... then one option I did see in some of the discussions on the problems of using moment with es6 (not specifically in the context of moment-timezone) was to explicitly add moment to the window object.

e.g.
```
import moment from 'moment/src/moment.js' (somehow...)
window.moment = moment;
import 'moment-timezone/build/moment-timezone-with-data.min.js'; 
```

But not sure if that would work if the amd/commonjs sniffing code is still screwing things up and preventing the browser global stuff from correctly working. Also, not sure if the above would have problems due to the aync nature of imports. But if it imports things correctly then you could try using dynamic imports to control the timing of things.

**Comment by awikkerink** [comment]
Maybe:

```javascript
# moment-global.js
import moment from 'moment/src/moment.js' (somehow...)
window.moment = moment;

# root.js
import './moment-global.js';
import 'moment-timezone/build/moment-timezone-with-data.min.js';
```

**Comment by mdulat** [comment]
@mdgbayly @awikkerink all those variations lead to the same console error at the same location, where `root` is undefined.

**Comment by awikkerink** [comment]
Finally got the Galen tests running locally: 

[reports.zip](https://github.com/BrightspaceUI/datetime-picker/files/2738197/reports.zip)

File: test/acceptance/galen.test.js [hunk 3]
```diff
@@ -66,7 +65,6 @@ polymerTests(browsers, function(test) {
 		shadow && queryParams.push('dom=shadow');
 		mobile && queryParams.push('width=280px');
 
-		var testEndpoint = mainline ? mainlineEndpoint : oneDotXEndpoint;
```
Comment (awikkerink) [inline test/acceptance/galen.test.js]: `testEndpoint` is used here: https://github.com/BrightspaceUI/datetime-picker/pull/53/files#diff-08507c6e4fae428cfce497780abe48a8R96, but isn't defined. I think this is why the tests are failing. One of the request URLs for a failing test is: `http://localhost:8081/components/d2l-datetime-picker/demo/galen.html?dir=rtl?dir=rtl?dom=shadow`

Comment (mdulat) [reply test/acceptance/galen.test.js]: But it's defined at the top of the file (line 47)

Comment (awikkerink) [reply test/acceptance/galen.test.js]: I think the modifications to testEndpoint are persisting.

**Pull request merged**